harness/determined/tensorboard/metric_writers/pytorch.py

```diff
@@ -33,16 +48,20 @@ def __init__(self, context):
 
             def train_batch(self, batch, epoch_idx, batch_idx):
                 self.logger.writer.add_scalar('my_metric', np.random.random(), batch_idx)
+
+    ..warning::
+        TorchWriter() has been deprecated and will be removed in a future version.
+
+        Users are encouraged to switch to one of the following depending on use case:
+            - Trials Users, see ``determined.pytorch.PyTorchTrialContext.get_tensorboard_writer()``
+            - CoreAPI Users, create a ``torch.utils.tensorboard.SummaryWriter()`` object
+              and pass ``core_context.train.get_tensorboard_path()`` as the log_dir
```

Comment:
this text is fantastic, please also include it in your runtime warning as well.

Also, I suspect your `..warning::` needs a space `.. warning::` and I suspect that you would need a newline after `.. warning::` as well for sphinx syntax to be happy.